### PR TITLE
ci: Add bump release pipeline

### PR DESCRIPTION
**Pull request by jcvenegas**
Add pipeline to bump repos in Kata

Signed-off-by: Jose Carlos Venegas Munoz <jose.carlos.venegas.munoz@intel.com>

### CODE REVIEW

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  }
+  environment {
+    PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/go/bin"
+    BRANCH="${params.BRANCH}"
+    NEW_VERSION="${params.NEW_VERSION}"
+  }
+  stages {
+    stage('Bump repos') {
+    // FIXME need better label for ubuntu, generic
+      agent { label 'ubuntu-1710' }
```
Comment (jcvenegas) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: @chavafg  is there  just an ubuntu label?

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+          export GOPATH="${WORKSPACE}/go"
+          export GOROOT="/usr/local/go"
+          export CI="true"
+          export tests_repo="github.com/kata-containers/tests"
+          export tests_repo_dir="${GOPATH}/src/${tests_repo}"
+          export packaging_repo="github.com/kata-containers/packaging"
+          export packaging_repo_dir="${GOPATH}/src/${packaging_repo}"
+          echo 'export PATH=$PATH:/usr/local/go/bin' | sudo tee -a /etc/profile
+          export PATH=$PATH:/usr/local/go/bin
+          export GITHUB_TOKEN=$GITHUB_TOKEN
```
Comment (jcvenegas) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: We should send this to null

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+          mkdir -p "${GOPATH}"
+          mkdir -p $(dirname "${tests_repo_dir}")
+          mkdir -p $(dirname "${kata_repo_dir}")
+          [ -d "${tests_repo_dir}" ] || git clone "https://${tests_repo}.git" "${tests_repo_dir}"
```
Comment (jcvenegas) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: here we could use our scripts to install docker as long does not install kata

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+      description: "Kata Containers Branch"
+    )
+    string(
+      name: 'NEW_VERSION',
+      defaultValue:"",
+      description: "Kata Containers version"
+    )
+  }
+  environment {
+    PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/usr/local/go/bin"
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: Nit: We probably don't need `/usr/local/games` :smile: :video_game: :space_invader:

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    BRANCH="${params.BRANCH}"
+    NEW_VERSION="${params.NEW_VERSION}"
+  }
+  stages {
+    stage('Bump repos') {
+    // FIXME need better label for ubuntu, generic
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: FIXME - can you either resolve or create an issue link?

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      agent { label 'ubuntu-1710' }
+      steps {
+      // Todo
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: ?

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+        withCredentials([string(credentialsId: 'TODO_NEED_CREDENTIAL_ID', variable: 'GITHUB_TOKEN')]) {
+          sh '''
+          set +x
+          #This is needed to setup github credentials to do push
+	  # We dont want to log this so send it to null
+          (
+        	set +x
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: Can a `JenkinsFile` reference an external script? I think that would be clearer if possible.

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+				  export GITHUB_TOKEN=$GITHUB_TOKEN
+				  cat << EOT > /tmp/token.sh
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: I appreciate this is for a CI system. However, call me paranoid, but this is using a fixed filename in a world-writable directory to store security details. I think we need to do atleast the following (but feel free to add more paranoia ;) ...

- add a `trap` to remove the file.
- use `mktemp` to use a random filename with restricted permissions.

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+#!/bin/bash
+echo "\$GITHUB_TOKEN"
+#esac
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: Commented out code.

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+          "${GOPATH}/src/${tests_repo}/.ci/install_go.sh" -p -f
+          go version
+          sudo -E apt-get -y install apt-transport-https ca-certificates software-properties-common
+          curl -sL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
+          arch=$(dpkg --print-architecture)
+          sudo -E add-apt-repository "deb [arch=${arch}] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable"
+          sudo -E apt-get update
+          sudo -E apt-get -y install docker-ce
+          [ -d "${packaging_repo_dir}" ] || git clone "https://${packaging_repo}.git" "${packaging_repo_dir}"
+          #[ -d "${packaging_repo_dir}" ] || git clone "https://github.com/jcvenegas/packaging-1.git" "${packaging_repo_dir}"
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: Commented out code.

File: Jenkinsfiles/release_pieline/Jenkinsfile [hunk 11]
```diff
@@ -0,0 +1,106 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+pipeline {
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/Jenkinsfile]: Ooooh - is this part of https://github.com/kata-containers/ci/issues/15 ? :)

**Comment by jcvenegas** [comment]
@jodh-intel the Jenkinsfile can not reference scripts related with it, but I factor out the bash code to scripts to make it easier to read.

File: Jenkinsfiles/release_pieline/git_credential_cache.sh [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+die() {
+	echo >&2 "ERROR: $*"
+	exit 1
+}
+
+init_git_credential_cache() {
+	#This is needed to setup github credentials to do push  in a job
+	(
+
```
Comment (jodh-intel) [inline Jenkinsfiles/release_pieline/git_credential_cache.sh]: Nit: extra blank line.

**Pull request merged**